src/frontend/src/routes/(new-styling)/manage/(authenticated)/+layout.svelte

```diff
@@ -205,71 +202,70 @@
       </Button>
     </header>
     <!-- Page content -->
-    <main
-      class={[
-        "flex flex-col px-4 py-5 sm:px-8 sm:py-3 md:px-12",
-        "transition-opacity duration-200",
-        isSwitchingIdentity && "pointer-events-none opacity-0",
-      ]}
-    >
+    <main class="flex flex-col px-4 py-5 sm:px-8 sm:py-3 md:px-12">
       {@render children()}
     </main>
-    {#if isSwitchingIdentity}
-      <div
-        class="absolute top-1/2 left-1/2 -translate-1/2"
-        transition:fade={{ duration: 200 }}
-      >
-        <ProgressRing class="text-fg-tertiary size-14" />
-      </div>
-    {/if}
     <div class="h-[env(safe-area-inset-bottom)]"></div>
   </div>
 </div>
 
 {#if isIdentityPopoverOpen}
   <Popover
     anchor={identityButtonRef}
-    onClose={() => (isIdentityPopoverOpen = false)}
+    onClose={() => {
+      if (isAuthenticating) {
+        return;
+      }
+      isIdentityPopoverOpen = false;
+    }}
     direction="down"
     align="end"
     distance="0.75rem"
   >
     <IdentitySwitcher
       selected={$authenticatedStore.identityNumber}
       identities={lastUsedIdentities}
-      switchIdentity={handleSwitchIdentity}
-      useAnotherIdentity={() => {
+      onSwitchIdentity={handleSignIn}
+      onUseAnotherIdentity={() => {
         isIdentityPopoverOpen = false;
         isAuthDialogOpen = true;
       }}
+      onError={(error) => {
+        isIdentityPopoverOpen = false;
+        isAuthenticating = false;
+        handleError(error);
```

Comment:
This seems like a suboptimal way to write the error handler. 

How does the reader supposed to remember what happens _here_ vs. `handleError`? 

Each UI component should clearly define its state machine, and resetting it should be clearly defined in one place (as returning to the initial state, not an arbitrary sequence of low-level operations, because this is prone to errors). 

Since this isn't immediately actionable, NCR for this PR.